src/lib/device/device.h

```diff
@@ -159,12 +159,16 @@ class DeviceManager {
 
     const Chunk* get_chunk(uint32_t chunk_id) const {
         std::unique_lock lg{m_vdev_mutex};
-        return (chunk_id == INVALID_CHUNK_ID) ? nullptr : m_chunks[chunk_id].get();
+        // if a pdev is misssing when restart, chunk_id from cleint might be larger than m_chunks.size()
+        if (chunk_id == INVALID_CHUNK_ID || chunk_id >= m_chunks.size()) return nullptr;
```

Comment:
we probabaly need to change the `m_chunks` to a map, or we need to insert dummy entries.

As the missing chunks can be in the middle , not the tail.  E.G.  chunk 0/1 from disk 1, 2/3 from disk 2, 4/5 from disk 3, 6/7 from disk 4

we might lost disk 2,  in your current implementation ,    we probably get 4/5 when requesting 2/3

Reply:
m_chunks is actually sparse vector, which will insert dummy element up to the index. it means if we have a sparse vector with a size of 10  and we use sparse_vector[20] to insert a new element, it will fill up [10-19] with elements by the default constructor.
```
sisl::sparse_vector< shared< Chunk > > m_chunks; 
```
pls refer to https://github.com/eBay/sisl/blob/8808be84afd8a967f83feb437e3652a87bf13b46/include/sisl/fds/sparse_vector.hpp#L62